### PR TITLE
HTML `<search>` element

### LINKED ISSUE #385: Document the <search> element
### Proposal

https://html.spec.whatwg.org/multipage/grouping-content.html#the-search-element

https://github.com/whatwg/html/pull/7320

### Task list

_No response_

### Additional information

https://github.com/whatwg/html/pull/7320 contains links to various relevant browser bugs. This isn't implemented anywhere yet but probably will be soon since it's simple.

### Are you willing to support this work?

I can help review

### Depends on

_No response_

### LINKED ISSUE #26006: Add new `<search>` element to docs
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/search

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

There's now a new `<search>` element in the [HTML spec](https://html.spec.whatwg.org/multipage/grouping-content.html#the-search-element), created about two weeks ago, which is not yet included in these docs.

### What did you expect to see?

Documentation about the element

### Do you have any supporting links, references, or citations?

* https://html.spec.whatwg.org/multipage/grouping-content.html#the-search-element
* https://github.com/whatwg/html/commit/c598ff023f081dd3f03b2e43177a632fb7dc92ec

### Do you have anything more you want to share?

This new element should also be mentioned on the [`<input type="search">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search), [`role="search"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/search_role) and [`role="searchbox"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/searchbox_role) pages.

### PR DESCRIPTION
**Pull request by ramiy**
### Description

On March 2023, a new HTML `<search>` element was [added to the HTML specification](https://html.spec.whatwg.org/multipage/grouping-content.html#the-search-element).

Fix https://github.com/mdn/content/issues/26006
Fix https://github.com/mdn/mdn/issues/385

### Related issues

* Safari: https://bugs.webkit.org/show_bug.cgi?id=254327
* Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1824121
* Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1294294, https://bugs.chromium.org/p/chromium/issues/detail?id=1277435

### CODE REVIEW

**Comment by Josh-Cena** [comment]
cc @domenic

**Comment by ramiy** [comment]
Markdown lint - Multiple headings with the same content [Context: `### Result`]

In other HTML element articles there are multiple headings with the same content, [see the header tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header).

**Comment by Josh-Cena** [comment]
@ramiy The typical structure is to use H4 headings.

```md
## Examples

### Doing something

#### HTML

#### Result

### Doing something else

#### HTML

#### Result
```

File: files/en-us/web/html/element/search/index.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+</header>
+```
+
+### Result
+
+{{EmbedLiveSample('Header search form')}}
+
+### Web app search
+
+JavaScript search functionality in a web application. No form element to perform a server-side submission, but for semantics a search element represents search and filtering capabilities.
```
Comment (estelle) [inline files/en-us/web/html/element/search/index.md:45]:
```suggestion
This example demonstrates potential DOM content when dynamically including JavaScript search functionality in a web application. When search functionality is implemented entirely with JavaScript, if no form is submitted, neither a {{HTMLElement("form")}} element nor a submit {{HTMLElement("button")}} is required. For semantics, the `<search>` element is included to contain the search and filtering capabilities.
```

File: files/en-us/web/html/element/search/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      <!-- search result content -->
+    </ul>
+    <output id="no-results">
+      <!-- no results content -->
+    </output>
+  </section>
+</search>
+```
+
+### Result
```
Comment (estelle) [inline files/en-us/web/html/element/search/index.md:70]:
```suggestion
#### Result
```

File: files/en-us/web/html/element/search/index.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+{{EmbedLiveSample('Web app search')}}
```
Comment (estelle) [inline files/en-us/web/html/element/search/index.md:72]:
```suggestion
{{EmbedLiveSample('Web app search')}}

> **Note:** Remember that some users don't have JavaScript, and none of your users have JavaScript running until the JavaScript is successfully downloaded, parsed, and executed, ensure your users can access the content of your site with JavaScript disabled.
```

File: files/en-us/web/html/element/search/index.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+### Multiple searches
+
+A page with two search features. The first is a global site search located on the header. The second is a search and filter based on the page context, in our example a car search.
```
Comment (estelle) [inline files/en-us/web/html/element/search/index.md:76]:
```suggestion
This example demonstrates a page with two search features. The first is a global site search located on the header. The second is a search and filter based on the page context, in our example a car search.
```

File: files/en-us/web/html/element/search/index.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+       ...
+     </search>
+     <article>
+      <!-- search result content -->
+    </article>
+  </main>
+</body>
+```
+
+### Result
```
Comment (estelle) [inline files/en-us/web/html/element/search/index.md:99]:
```suggestion
#### Result
```

**Review by estelle — CHANGES_REQUESTED** [review]
Thank you for contributing this `<search>` element content. i made a few suggestions, and a fix for the prettier issues.

**Review by estelle — APPROVED** [review]
Thanks! 🎉

**Pull request merged**